cmds/db-manager/migration/migrate.go

```diff
@@ -157,25 +173,37 @@ func migrate(cmd *cobra.Command, _ []string) error {
 
 		// Ensure SQL session has implicit transactions disabled for CRDB versions 22.2+
 		sessionConfigurationSQL := ""
-		if ds.Version.SemVer.Compare(*semver.New("22.2.0")) >= 0 {
+		if isCockroach && ds.Version.SemVer.Compare(*semver.New("22.2.0")) >= 0 {
 			sessionConfigurationSQL = "SET enable_implicit_transaction_for_batch_statements = false;\n"
 		}
 
-		migrationSQL := sessionConfigurationSQL + fmt.Sprintf("USE %s;\n", dbName) + string(rawMigrationSQL)
+		migrationSQL := ""
+		if isCockroach {
+			migrationSQL = sessionConfigurationSQL + fmt.Sprintf("USE %s;\n", dbName) + string(rawMigrationSQL)
+		} else {
+			ds, err = datastore.Dial(ctx, connectParameters)
+			if err != nil {
+				return fmt.Errorf("failed to reconnect to database %s: %w", dbName, err)
+			}
+			migrationSQL = sessionConfigurationSQL + string(rawMigrationSQL)
+		}
```

Comment:
While functionally equivalent, shouldn't this be inside a `if isYugabyte` block to be conceptually correct? IIUC this is required for yugabyte to reconnect to a different DB, this is not required because 'the DB is not cockroach'.